tests/run-make/staticlib-thin-archive/rmake.rs

```diff
@@ -0,0 +1,28 @@
+// Regression test for https://github.com/rust-lang/rust/issues/107407 which
+// checks that rustc can read thin archive. Before the object crate added thin
+// archive support rustc would add emit object files to the staticlib and after
+// the object crate added thin archive support it would previously crash the
+// compiler due to a missing special case for thin archive members.
+use std::path::Path;
+
+use run_make_support::{llvm_ar, rust_lib_name, rustc, static_lib_name};
+
+fn main() {
+    std::fs::create_dir("archive").unwrap();
```

Comment:
This could be `rfs::create_dir("archive")`, which automatically unwraps and has a nice error message. (add `rfs` to run_make_support imports)

Reply:
Done